sympy/tensor/array/array_derivatives.py

```diff
@@ -76,6 +76,9 @@ def _call_derive_default(expr: Expr, v: Expr) -> Expr | None:
         if expr.has(v):
             return _matrix_derivative(expr, v)
         else:
+            if isinstance(expr, MatrixExpr) and expr.shape == (1, 1):
+                if isinstance(v, MatrixExpr):
+                    return ZeroMatrix(*v.shape)
```

Comment:
I believe this should be fixed in `array_derive` function. Matrix is converted to N-dim array, derived and then converted back to matrix form. The core of the matrix derivative algorithm is implemented in `sympy.tensor.array.expressions`.